pkg/cache/internal/informers_map.go

```diff
@@ -175,27 +189,42 @@ func (ip *specificInformersMap) waitForStarted(ctx context.Context) bool {
 }
 
 // HasSyncedFuncs returns all the HasSynced functions for the informers in this map.
-func (ip *specificInformersMap) HasSyncedFuncs() []cache.InformerSynced {
+func (ip *InformersMap) HasSyncedFuncs() []cache.InformerSynced {
 	ip.mu.RLock()
 	defer ip.mu.RUnlock()
-	syncedFuncs := make([]cache.InformerSynced, 0, len(ip.informersByGVK))
-	for _, informer := range ip.informersByGVK {
-		syncedFuncs = append(syncedFuncs, informer.Informer.HasSynced)
+	syncedFuncs := make([]cache.InformerSynced, 0, len(ip.informers.Structured))
```

Comment:
nit: should we use len(ip.informers.Structured) + len(ip.informers.Unstructured) + len (ip.informers.Metadata) instead?